Source/inv.cpp

```diff
@@ -1932,27 +1933,16 @@ bool UseScroll()
 	if (pcurs != CURSOR_HAND)
 		return false;
 
-	auto &myPlayer = Players[MyPlayerId];
+	Player &myPlayer = Players[MyPlayerId];

```

Comment:
Why did you change from `auto` to `Player` here?

Reply:
On `auto`, from https://google.github.io/styleguide/cppguide.html#Type_deduction:

> Use type deduction only if it makes the code clearer to readers who aren't familiar with the project, or if it makes the code safer. Do not use it merely to avoid the inconvenience of writing an explicit type.

Reply:
Well that's Google's opinion right? Are we following _that_ standard for `auto`? I don't think so.

@AJenbo , can we enforce proper use of `auto` to avoid these discussions? Current codebase doesn't follow the rule @glebm just shared.

Reply:
It may not make the code clearer, but I also don't think it makes it any less clear. I would prefer `auto` here simply for consistency. All the cleanups recently have been using it in this same way.

Reply:
> All the cleanups recently have been using it in this same way.

That's the most important aspect of my question: we've been doing tons of cleanup that includes `auto` uses like the one here, and now we are undoing some of it.

If we keep doing and undoing it that would be really confusing.